csharp/src/Microsoft.ML.OnnxRuntime/Tensors/CompressedSparseTensor.cs

```diff
@@ -0,0 +1,528 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+// This file is copied and adapted from the following git repository -
+// https://github.com/dotnet/corefx
+// Commit ID: bdd0814360d4c3a58860919f292a306242f27da1
+// Path: /src/System.Numerics.Tensors/src/System/Numerics/Tensors/CompressedSparseTensor.cs
+// Original license statement below -
+
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System.Diagnostics;
+using System.Linq;
+using System;
+
+namespace Microsoft.ML.OnnxRuntime.Tensors
+{
+    /// <summary>
+    /// Represents a tensor using compressed sparse format
+    /// For a two dimensional tensor this is referred to as compressed sparse row (CSR, CRS, Yale), compressed sparse column (CSC, CCS)
+    /// 
+    /// In this format, data that is in the same value for the compressed dimension has locality
+    /// 
+    /// In standard layout of a dense tensor, data with the same value for first dimensions has locality.
+    /// As such we'll use reverseStride = false (default) to mean that the first dimension is compressed (CSR)
+    /// and reverseStride = true to mean that the last dimension is compressed (CSC)
+    /// 
+    /// </summary>
+    /// <typeparam name="T"></typeparam>
+    internal class CompressedSparseTensor<T> : Tensor<T>
```

Comment:
Do you plan on using this class? I think it may make sense to not bring it until you need it.